### PR TITLE
fix: parse signed CTO in v1 trun boxes

### LINKED ISSUE #332: Garbled CEA captions
In https://github.com/google/shaka-player/issues/2395, we received a report of garbled CEA captions in Shaka Player.  We do not know what is causing it, but we can reproduce the issue with mux.js in a standalone node script which is very similar in structure to how we use mux.js in Shaka Player:

```js
const muxjs = require('mux.js');
const fs = require('fs');

const CaptionParser = class {
  constructor() {
    this.muxCaptionParser_ = new muxjs.mp4.CaptionParser();
    this.videoTrackIds_ = [];
    this.timescales_ = {};
  }

  parseInitSegment(data) {
    this.videoTrackIds_ = muxjs.mp4.probe.videoTrackIds(data);
    this.timescales_ = muxjs.mp4.probe.timescale(data);
    this.muxCaptionParser_.init();
  }

  parseMediaSegment(data) {
    const parsed = this.muxCaptionParser_.parse(
        data, this.videoTrackIds_, this.timescales_);
    const captions = parsed && parsed.captions ? parsed.captions : [];
    this.muxCaptionParser_.clearParsedCaptions();
    return captions;
  }
};

function readFile(path) {
  return new Uint8Array(fs.readFileSync(path));
}

// argv[0] is the name of the interpreter
// argv[1] is the name of this script
if (process.argv.length < 4) {
  console.log('Usage: ' + process.argv[0] + ' ' + process.argv[1] +
              '<INIT_SEGMENT> <MEDIA_SEGMENT> [<MEDIA_SEGMENT> ...]');
  process.exit(0);
}

const initSegmentPath = process.argv[2];
const mediaSegmentPaths = process.argv.slice(3);

const initSegment = readFile(initSegmentPath);
console.log('Init segment:', initSegmentPath, initSegment.length + ' bytes');

const p = new CaptionParser();
p.parseInitSegment(initSegment);

for (const path of mediaSegmentPaths) {
  const segment = readFile(path);
  console.log('Media segment:', path, segment.length + ' bytes');

  for (const caption of p.parseMediaSegment(segment)) {
    console.log(caption);
  }
}
```

The output is:

```
{ startPts: 4296348676,
  endPts: 4296519847,
  text: 'e  iuc\nri Mm,eneadiouR-- -- <i>Hetageuseu</i>',
  stream: 'CC1',
  startTime: 47737.20751111111,
  endTime: 47739.10941111111 }
```

That text is supposed to be English, though I don't have a working parser for comparison to say exactly what that particular piece of text is meant to be.  We get the same results for both encrypted and clear versions of the content, so we know that the encryption is not being applied to the CC parts of the segment.

I've asked permission to share the init segment and one encrypted media segment with you, and I will follow up with those as soon as I have permission.

### PR DESCRIPTION
**Pull request by joeyparrish**
This resolves a TODO in mp4-inspector.js about v1 trun boxes.

Closes #332

### CODE REVIEW

**Comment by joeyparrish** [comment]
@ppatlolla-turner, @gkatsev

**Comment by gkatsev** [comment]
Thanks for tracking this down, @joeyparrish!

File: lib/tools/mp4-inspector.js [hunk 1]
```diff
@@ -727,8 +727,11 @@ var
           offset += 4;
         }
         if (sampleCompositionTimeOffsetPresent) {
-          // Note: this should be a signed int if version is 1
-          sample.compositionTimeOffset = view.getUint32(offset);
+          if (result.version == 1) {
```
Comment (gkatsev) [inline lib/tools/mp4-inspector.js:730]:
```suggestion
          if (result.version === 1) {
```

File: lib/tools/mp4-inspector.js [hunk 2]
```diff
@@ -750,8 +753,11 @@ var
...
```
Comment (gkatsev) [inline lib/tools/mp4-inspector.js:756]:
```suggestion
          if (result.version === 1) {
```

**Comment by joeyparrish** [comment]
Thanks for pointing that out.  I'm not that familiar with the project's conventions.

**Comment by gkatsev** [comment]
I can confirm that Is see English output via the example you posted in #332.

the output is
```sh
$ node index.js CEA_segments/init_segment.mp4 CEA_segments/encrypted_media_segment.mp4
Init segment: CEA_segments/init_segment.mp4 1957 bytes
Media segment: CEA_segments/encrypted_media_segment.mp4 156915 bytes
{ startPts: 1381380,
  endPts: 1552551,
  text: '<i>Heritage Museum,</i>',
  stream: 'CC1',
  startTime: 15.348666666666666,
  endTime: 17.250566666666668 }
```

**Comment by joeyparrish** [comment]
Thanks!  Yeah, I tested with a few more segments as well, and the whole first scene of this video is working correctly with the fix.

**Comment by gkatsev** [comment]
Really appreciate your help on this @joeyparrish

**Comment by joeyparrish** [comment]
I can't take much credit.  It was an easy fix.  The insight came from a coworker.

**Pull request merged**